temporal-sdk/src/main/java/io/temporal/internal/client/RootScheduleClientInvoker.java

```diff
@@ -0,0 +1,255 @@
+import io.temporal.internal.client.external.GenericWorkflowClient;
+import io.temporal.internal.common.ProtobufTimeUtils;
+import io.temporal.internal.common.SearchAttributesUtil;
+import java.util.*;
+import java.util.stream.StreamSupport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RootScheduleClientInvoker implements ScheduleClientCallsInterceptor {
+  private static final Logger log = LoggerFactory.getLogger(RootScheduleClientInvoker.class);
+
+  private final GenericWorkflowClient genericClient;
+
+  private final ScheduleClientOptions clientOptions;
+
+  private final ScheduleProtoUtil scheduleRequestHeader;
+
+  public RootScheduleClientInvoker(
+      GenericWorkflowClient genericClient, ScheduleClientOptions clientOptions) {
+    this.genericClient = genericClient;
+    this.clientOptions = clientOptions;
+    this.scheduleRequestHeader = new ScheduleProtoUtil(genericClient, clientOptions);
+  }
+
+  @Override
+  public void createSchedule(CreateScheduleInput input) {
+
+    CreateScheduleRequest.Builder request =
+        CreateScheduleRequest.newBuilder()
+            .setIdentity(clientOptions.getIdentity())
+            .setNamespace(clientOptions.getNamespace())
+            .setRequestId(UUID.randomUUID().toString())
+            .setScheduleId(input.getId())
+            .setSchedule(scheduleRequestHeader.scheduleToProto(input.getSchedule()));
+
+    if (input.getOptions().getMemo() != null) {
+      request.setMemo(
+          Memo.newBuilder()
+              .putAllFields(
+                  intoPayloadMap(clientOptions.getDataConverter(), input.getOptions().getMemo())));
+    }
+
+    if (input.getOptions().getSearchAttributes() != null
+        && !input.getOptions().getSearchAttributes().isEmpty()) {
+      request.setSearchAttributes(
+          SearchAttributesUtil.encode(input.getOptions().getSearchAttributes()));
+    }
+
+    if (input.getOptions().isTriggerImmediately()
+        || (input.getOptions().getBackfills() != null
+            && input.getOptions().getBackfills().size() > 0)) {
+      SchedulePatch.Builder patchBuilder = SchedulePatch.newBuilder();
+
+      if (input.getOptions().getBackfills() != null) {
+        input.getOptions().getBackfills().stream()
+            .forEach(b -> patchBuilder.addBackfillRequest(backfillToProto(b)));
+      }
+
+      if (input.getOptions().isTriggerImmediately()) {
+        patchBuilder.setTriggerImmediately(
+            TriggerImmediatelyRequest.newBuilder()
+                .setOverlapPolicy(input.getSchedule().getPolicy().getOverlap())
+                .build());
+      }
+
+      request.setInitialPatch(patchBuilder.build());
+    }
+
+    try {
+      genericClient.createSchedule(request.build());
+    } catch (Exception e) {
+      if (e instanceof StatusRuntimeException) {
+        StatusRuntimeException sre = (StatusRuntimeException) e;
+        if (Status.Code.ALREADY_EXISTS.equals(sre.getStatus().getCode())) {
```

Comment:
Isn't there a risk that StatusRuntimeException for any other codes than "ALREADY_EXISTS" are going to be swallowed here?

Reply:
Or just non-status-runtime exceptions, yes. I think this is a bug. I have opened #1817. Thanks!